src/services/ngx-uploader.ts

```diff
@@ -30,9 +30,15 @@ export class NgUploaderService {
 
   uploadFile(file: File): void {
     let xhr = new XMLHttpRequest();
-    let payload: FormData | File;
-
-    if (this.opts.multipart) {
+    let payload: FormData | File | {};
+    
+    if (this.opts.plainJson) {
+      let plainObj = {};
```

Comment:
is this needed? why you don't set `payload = JSON.stringify(this.opts.data)` directly?